tls/types_test.go

```diff
@@ -75,3 +81,20 @@ func TestDigitallySignedString(t *testing.T) {
 		}
 	}
 }
+
+func TestSignatureAlgorithm(t *testing.T) {
+	for _, test := range []struct {
+		name string
+		key  crypto.PublicKey
+		want SignatureAlgorithm
+	}{
+		{name: "ECDSA", key: new(ecdsa.PublicKey), want: ECDSA},
+		{name: "RSA", key: new(rsa.PublicKey), want: RSA},
+		{name: "DSA", key: new(dsa.PublicKey), want: DSA},
```

Comment:
Add a case for something that isn't ECDSA, RSA or DSA?

Reply:
Done